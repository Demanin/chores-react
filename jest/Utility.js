import crypto from 'crypto';

const getRandomString = (length = 20) => {
  return crypto.randomBytes(length).toString('hex');
};

const getRandomNumber = (min = 0, max = 9999999) => {
  min = Math.ceil(min);
  max = Math.floor(max);

  return Math.floor(Math.random() * (max - min)) + min;
};

const getRandomBool = () => {
  return !!getRandomNumber(0, 1);
};

export default {
  getRandomString,
  getRandomNumber,
  getRandomBool,
};
