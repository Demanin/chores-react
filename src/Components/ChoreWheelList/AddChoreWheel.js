import React, { Component } from 'react';
import { Button as BsButton } from 'react-bootstrap';

class AddChoreWheel extends Component
{
  render() {
    return (
      <div>
        <BsButton variant="success"
          onClick={this.props.onClick}
        >
          Add Chore Wheel
        </BsButton>
        <hr />
      </div>
    );
  }
}

export default AddChoreWheel;
