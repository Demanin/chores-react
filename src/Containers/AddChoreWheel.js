import BaseAddChoreWheel from 'src/Components/ChoreWheelList/AddChoreWheel';
import addChoreWheel from 'src/Actions/addChoreWheel';
import { connect } from 'react-redux';

const mapStateToProps = (state, ownProps) => {
  return {
    visibilityFilter: state.visibilityFilter,
  };
};

const mapDispatchToProps = (dispatch, ownProps) => {
  return {
    onClick: (isVisible) => {
      dispatch(addChoreWheel(1, isVisible));
    },
  };
};

const mergeProps = (stateProps, dispatchProps, ownProps) => {
  return {
    ...ownProps,
    ...stateProps,
    ...dispatchProps,
    onClick: () => {
      if ('SHOW_HIDDEN' === stateProps.visibilityFilter) {
        dispatchProps.onClick(false);

        return;
      }

      dispatchProps.onClick(true);
    },
  };
};

const AddChoreWheel = connect(
  mapStateToProps,
  mapDispatchToProps,
  mergeProps
)(BaseAddChoreWheel);

export default AddChoreWheel;
