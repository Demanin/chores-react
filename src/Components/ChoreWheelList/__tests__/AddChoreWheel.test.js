import AddChoreWheel from 'src/Components/ChoreWheelList/AddChoreWheel';
import React from 'react';
import Utility from 'jest/Utility';
import { shallow } from 'enzyme';

describe('AddChoreWheel', () => {
  let onClick;

  beforeEach(() => {
    onClick = jest.fn();
  });

  test('snapshot', () => {
    const fixture = shallow(<AddChoreWheel
      {...{ onClick }}
    />);

    expect(fixture).toMatchSnapshot();
  });

  test('single turn complete click', () => {
    const fixture = shallow(<AddChoreWheel
      {...{ onClick }}
    />);

    fixture.find('Button').simulate('click');

    expect(onClick).toHaveBeenCalled();
  });
});
