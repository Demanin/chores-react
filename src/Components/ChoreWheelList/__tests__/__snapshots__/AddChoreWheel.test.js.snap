// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddChoreWheel snapshot 1`] = `ShallowWrapper {}`;
